app/src/main/java/org/mozilla/fenix/settings/DefaultBrowserSettingsFragment.kt

```diff
@@ -89,6 +89,8 @@ class DefaultBrowserSettingsFragment : PreferenceFragmentCompat() {
             isChecked = context.settings().openLinksInAPrivateTab
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
+
+        DefaultBrowserPreference(requireContext()).unsetOpenLinksInAPrivateTabIfNecessary()
```

Comment:
I think this needs to be called before `findPreference<CheckBox...` so that `openLinksInAPrivateTab` has the latest value.